### PR TITLE
Mark Windows npipe support as stable (with a fix)

### PR DESCRIPTION
**Pull request by bsideup**
There was a bug in `FramedResponseStreamHandler`: if buffer has two lines, only first one was processed. Fixed by manually chunking the buffer

### CODE REVIEW

File: core/build.gradle [hunk 1]
```diff
@@ -36,7 +36,9 @@ shadowJar {
         'META-INF/services/org.jvnet.hk2.external.generator.*',
         'META-INF/services/org.glassfish.jersey.internal.spi.*',
         'META-INF/services/java.security.Provider',
+        'META-INF/services/java.sql.Driver',
```
Comment (kiview) [inline core/build.gradle]: Oh, I think we can remove this now also?

File: core/src/main/java/org/testcontainers/dockerclient/transport/okhttp/NamedPipeSocketFactory.java [hunk 2]
```diff
@@ -1,57 +1,123 @@
...
+
+            public InputStream getInputStream() {
+                return is;
+            }
+
+            public OutputStream getOutputStream() {
+                return os;
+            }
+
+            public void setTcpNoDelay(boolean bool) {
```
Comment (kiview) [inline core/src/main/java/org/testcontainers/dockerclient/transport/okhttp/NamedPipeSocketFactory.java]: Why do we need non `@Override` dummy methods?

File: core/src/main/java/org/testcontainers/dockerclient/transport/okhttp/OkHttpInvocationBuilder.java [hunk 3]
```diff
@@ -311,7 +311,13 @@ public void run() {
                     while (!(shouldStop.get() || source.exhausted())) {
                         int bytesReceived = inputStream.read(buffer);
 
-                        handler.channelRead(null, Unpooled.wrappedBuffer(buffer, 0, bytesReceived));
+                        int offset = 0;
+                        for (int i = 0; i < bytesReceived; i++) {
+                            if (buffer[i] == '\n' || i == bytesReceived - 1) {
```
Comment (kiview) [inline core/src/main/java/org/testcontainers/dockerclient/transport/okhttp/OkHttpInvocationBuilder.java]: Let's just extract those two boolean conditions into two variables, `lineBreak`  and `endOfBuffer`.  Maybe even adding a small comment that line break can happen, if the daemon is answering very fast and it means we got multiple responses.

File: core/src/main/java/org/testcontainers/dockerclient/transport/okhttp/OkHttpInvocationBuilder.java [hunk 4]
```diff
@@ -311,7 +311,16 @@ public void run() {
                     while (!(shouldStop.get() || source.exhausted())) {
                         int bytesReceived = inputStream.read(buffer);
 
-                        handler.channelRead(null, Unpooled.wrappedBuffer(buffer, 0, bytesReceived));
+                        int offset = 0;
+                        for (int i = 0; i < bytesReceived; i++) {
+                            // some handlers like JsonResponseCallbackHandler do not work with multi-line buffers
+                            boolean isLineBreak = buffer[i] == '\n';
```
Comment (gesellix) [inline core/src/main/java/org/testcontainers/dockerclient/transport/okhttp/OkHttpInvocationBuilder.java]: This one matches any line break, but I assume only line breaks between json objects should be considered (as delimiter for [line-delimited JSON](https://en.wikipedia.org/wiki/JSON_streaming#Line-delimited_JSON)). So, what happens when something like below is part of the response? 🤔 

```
{"foo":"bar\nbaz"}
{"anything":"else"}
```

Comment (bsideup) [reply core/src/main/java/org/testcontainers/dockerclient/transport/okhttp/OkHttpInvocationBuilder.java]: "\\n" != "\n" :)

Comment (gesellix) [reply core/src/main/java/org/testcontainers/dockerclient/transport/okhttp/OkHttpInvocationBuilder.java]: oh my. gotcha! :)

**Comment by rnorth** [comment]
Released for preview in [1.9.0-rc2](https://github.com/testcontainers/testcontainers-java/releases/tag/1.9.0-rc2), to be published on Bintray.

**Pull request merged**